frontend/src/app/supportAdmin/ManageFacility/ManageFacility.tsx

```diff
@@ -141,23 +161,26 @@ const ManageFacility = () => {
   }
 
   function handleClearFilter() {
-    updateLocalState(initialState);
+    orgSelectionRef.current?.clearSelection();
+    facilitySelectionRef.current?.clearSelection();
   }
 
   /**
    * Delete facility
    */
   const [deleteFacilityMutation] = useDeleteFacilityMutation();
   function handleDeleteFacility() {
-    deleteFacilityMutation({
-      variables: { facilityId: localState.facilityId },
-    }).then(() => {
-      showSuccess(
-        "",
-        `Facility ${localState.facility?.name} successfully deleted`
-      );
-      handleClearFilter();
-    });
+    if (localState.facilityId) {
```

Comment:
why is this condition needed? in theory the delete button will display only if a facility has been loaded